code/game/gamemodes/shadowling/shadowling.dm

```diff
@@ -0,0 +1,319 @@
+#define LIGHT_DAM_THRESHOLD 4
+#define LIGHT_HEAL_THRESHOLD 2
+#define LIGHT_DAMAGE_TAKEN 7
+
+/*
+
+SHADOWLING: A gamemode based on previously-run events
+
+Aliens called shadowlings are on the station.
+These shadowlings can 'enthrall' crew members and enslave them.
+They also burn in the light but heal rapidly whilst in the dark.
+The game will end under two conditions:
+	1. The shadowlings die
+	2. The emergency shuttle docks at CentCom
+
+Shadowling strengths:
+	- The dark
+	- Hard vacuum (They are not affected by it, but are affected by starlight!)
+	- Their thralls who are not harmed by the light
+	- Stealth
+
+Shadowling weaknesses:
+	- The light
+	- Fire
+	- Enemy numbers
+	- Burn-based weapons and items (flashbangs, lasers, etc.)
+
+Shadowlings start off disguised as normal crew members, and they only have two abilities: Hatch and Enthrall.
+They can still enthrall and perhaps complete their objectives in this form.
+Hatch will, after a short time, cast off the human disguise and assume the shadowling's true identity.
+They will then assume the normal shadowling form and gain their abilities.
+
+The shadowling will seem OP, and that's because it kinda is. Being restricted to the dark while being alone most of the time is extremely difficult and as such the shadowling needs powerful abilities.
+Made by Xhuis
+
+*/
+
+
+
+/*
+	GAMEMODE
+*/
+
+
+/datum/game_mode
+	var/list/datum/mind/shadows = list()
+	var/list/datum/mind/thralls = list()
+	var/list/shadow_objectives = list()
+	var/required_thralls = 15 //How many thralls are needed (hardcoded for now)
+	var/shadowling_ascended = 0 //If at least one shadowling has ascended
+	var/shadowling_dead = 0 //is shadowling kill
+	var/objective_explanation
+
+
+/proc/is_thrall(var/mob/living/M)
+	return istype(M) && M.mind && SSticker && SSticker.mode && (M.mind in SSticker.mode.thralls)
+
+
+/proc/is_shadow_or_thrall(var/mob/living/M)
+	return istype(M) && M.mind && SSticker && SSticker.mode && ((M.mind in SSticker.mode.thralls) || (M.mind in SSticker.mode.shadows))
+
+
+/proc/is_shadow(var/mob/living/M)
+	return istype(M) && M.mind && SSticker && SSticker.mode && (M.mind in SSticker.mode.shadows)
+
+
+/datum/game_mode/shadowling
+	name = "shadowling"
+	config_tag = "shadowling"
+	antag_flag = ROLE_SHADOWLING
+	//required_players = 30
```

Comment:
It has been, but due to this being closed, the latest commits don't show on the PR